src/main.rs

```diff
@@ -495,15 +495,12 @@ impl App {
         let config = self.build_stub_config(args)?;
 
         let stub_generator = match args.get_one::<PathBuf>("from-file") {
-            Some(fname) => {
-                if fname.to_str() == Some("-") {
-                    let mut input = String::new();
-                    std::io::Read::read_to_string(&mut std::io::stdin(), &mut input)?;
-                    input
-                } else {
-                    std::fs::read_to_string(fname)?
-                }
+            Some(fname) if fname.to_str() == Some("-") => {
+                let mut input = String::new();
+                std::io::Read::read_to_string(&mut std::io::stdin(), &mut input)?;
```

Comment:
This should probably print a message explaining that it's reading from `stdin` you need to send an `EOF` signal to see the result.

Reply:
Printing a message would only makes sense if you're not piping into `clash`. I think there's a way to check that, but the overlap of users who know about `-` as stdin and users who do not know about EOF is rather small. Not sure if it's worth the effort.